### PR TITLE
feat: add ButtonStyleTypes.Premium

### PR DESCRIPTION
**Pull request by goatslacker**
Relevant docs changes: https://github.com/discord/discord-api-docs/pull/6875

### CODE REVIEW

File: src/components.ts [hunk 1]
```diff
@@ -15,33 +15,44 @@ export enum MessageComponentTypes {
...
 export enum ButtonStyleTypes {
   PRIMARY = 1,
   SECONDARY = 2,
   SUCCESS = 3,
   DANGER = 4,
   LINK = 5,
+  PREMIUM = 6,
 }
 
+type PremiumButton = {
```
Comment (JustinBeckwith) [inline src/components.ts:27]: Any reason for having an entirely different type here?  Is it the same thing under the hood?

Comment (goatslacker) [reply src/components.ts:27]: yeah `sku_id` is required and should only be present if the button style is premium, we validate this in the API layer so I made a type that ensures it so devs don't forget to include it.

Comment (JustinBeckwith) [reply src/components.ts:27]: Cool!  Can we make Button an interface, and then have PremiumButton extend it?  So we don't replicate all the fields?

Comment (goatslacker) [reply src/components.ts]: done here: https://github.com/discord/discord-interactions-js/pull/55/commits/d346e7fd1dbdad0481810a06311ef7398309ebab

File: src/components.ts [hunk 2]
```diff
@@ -15,33 +15,43 @@ export enum MessageComponentTypes {
...
+  sku_id: string;
+  style: ButtonStyleTypes.PREMIUM;
+}
+
+interface LinkButton extends BaseButton {
+  url: string;
+  style: ButtonStyleTypes.LINK;
+}
+
+interface AllOtherButtons extends BaseButton {
```
Comment (JustinBeckwith) [inline src/components.ts:44]: Heh, this interface name doesn't feel quite right.  How about just `Button`?  Or if we really need to differentiate, `StandardButton`?

Comment (goatslacker) [reply src/components.ts:44]: hah yeah I don't like the name either. Button would be great but we export `Button` below. I'm fine with `StandardButton`

Comment (goatslacker) [reply src/components.ts:44]: how about `CustomButton`?

Comment (JustinBeckwith) [reply src/components.ts]: That works for me!

**Review by JustinBeckwith — APPROVED** [review]
LGTM, assuming all the backend stuff is actually sorted

**Pull request merged**